robot_controllers/src/cartesian_twist.cpp

```diff
@@ -302,15 +326,34 @@ void CartesianTwistController::command(const geometry_msgs::Twist::ConstPtr& goa
   ros::Time now(ros::Time::now());
   {
     boost::mutex::scoped_lock lock(mutex_);
+    twist_command_frame_ = frame;
     twist_command_ = twist;
     last_command_time_ = now;
   }
 
-  // Try to start up
-  if (manager_->requestStart(getName()) != 0)
+  if(frame=="")
   {
-    ROS_ERROR("CartesianTwistController: Cannot start, blocked by another controller.");
-    return;
+    if(frame_command_enabled_ == false)
+    {
+      last_frame_command_time_ = now;
+      frame_command_enabled_ = true;
+    }
+    if((now-last_frame_command_time_) > ros::Duration(0.5)&&stop_requested_==true)
+    {
+      manager_->requestStop(getName());
+      stop_requested_ = false;
+    }
+  }
+  else
+  {
+    frame_command_enabled_ = false;
+    stop_requested_ = true;
+    // Try to start up
+    if (manager_->requestStart(getName()) != 0)
+    {
+      ROS_ERROR("CartesianTwistController: Cannot start, blocked by another controller.");
+      return;
+    }
```

Comment:
I'm not sure I follow this logic -- if the frame_id is empty, we don't start. If the frame_id becomes empty later, we stop but only after 0.5 seconds? Why 0.5 seconds? Why not just ignore commands that have no frame_id, and then let the usual timeout take effect (which would remove the need for several variables, and a bunch of logic above)?